### ISSUE TITLE
Framer Motion `useScroll()` incompatibility: providing `wrapperElement` ref.

### ISSUE THREAD

**Issue by damnsamn** [studio-freight/react-lenis#35]
I'm bumping into an issue at the moment where I'm using Framer Motion's `useScroll()` hook, which takes a `container: RefObject<HTMLElement>` to track its scroll progress and map to animations etc. But because `<ReactLenis>` subverts the expected behaviour and returns a `lenis` instance instead of an element ref via `useImperitiveHandle`, it becomes difficult to use these libraries together.

Is there a way, or can there be a way, to preserve the `ref` passed to `<ReactLenis>` as an element ref and provide some other method of passing the `lenis` instance back up?

**Comment by clementroche** [comment]
you can use `lenis.options.wrapper`

**Comment by damnsamn** [comment]
Sorry - I'm referring more to passing through the provided ref unchanged. I'm aware that you can access the wrapper element directly - other react library will be generally expecting element **refs** via `useRef()`, as opposed to actual HTMLElement instances.

For example, using Framer Motion would generally look like this:
```
const scrollRef = useRef<HTMLDivElement>(null);
const { scrollY, scrollYProgress } = useScroll({ container: scrollRef });

return <div ref={scrollRef}>{children}</div>
```

But when that scrolling element is a `<ReactLenis>`, we're unable to provide the `container` option with an element `ref` **at all** (unless I'm missing something). For example, with Lenis:
```
const lenisRef = useRef<LenisInstance>(null);
const { scrollY, scrollYProgress } = useScroll({ container: lenisRef.options.wrapper });

return <ReactLenis ref={scrollRef}>{children}</ReactLenis>
```

The above snippet will not function, as it's expecting a RefObject to unwrap (as far as I can tell)

**Comment by damnsamn** [comment]
To be clear, I'm not asking anything necessarily specific to Framer Motion or expecting any accounting for that library specifically.

At the moment it's more a question of being able to pass `<ReactLenis>` a ref and have it fill that ref with its root rendered element - as opposed to exposing the lenis instance only via `useImperativeHandle(ref, ()=>lenis, [lenis])`.

**Comment by clementroche** [comment]
🤷‍♂️
```js
const scrollRef = useRef<LenisInstance>(null);
const { scrollY, scrollYProgress } = useScroll({ container: { current: scrollRef.options.wrapper } });

return <ReactLenis ref={scrollRef}>{children}</ReactLenis>
```

Also nothing is stopping you from storing as ref if needed

```js
const lenisRef = useRef()
const wrapperRef = useRef()
const contentRef = useRef()

<ReactLenis ref={(lenis)=>{
  if(lenis) {
    lenisRef.current = lenis
    wrapperRef.current = lenis.options.wrapper
    contentRef.current = lenis.options.content
  }
}}  />
```

In my opinion lenis is providing anything you need. It was designed to make it easier to instanciate with React so if you need something really specific you can still use vanilla Lenis. 
To change what's returned by useImperativeHandle will introduce a breaking change in the API, i prefer to avoid that.

**Comment by clementroche** [comment]
Also if you want to continue this discussion please provide a reproduction link with framer motion and react-lenis so i can figure it out from there

**Comment by damnsamn** [comment]
@clementroche 
In practice those solutions don't really work - they result in a hydration error within Framer Motion, presumably caused by a race condition due to the ref not being filled when it typically would (provided directly to a dom element).

I've reproduced this issue in the following repo: https://github.com/damnsamn/bug-lenis-framer-motion

I appreciate your help on this issue mate - you guys are doing awesome work

**Comment by damnsamn** [comment]
In the mean time - as you say I think I might be better served writing my own react wrapper for vanilla Lenis for my current project.

**Comment by clementroche** [comment]
Can you share a codesandbox instead of a github repository ?.

**Comment by damnsamn** [comment]
@clementroche 
Sure
https://codesandbox.io/s/busy-aj-csx2qg?file=/src/App.tsx

**Comment by clementroche** [comment]
well @damnsamn it works 🤷‍♂️

**Comment by damnsamn** [comment]
Sorry did you read the comments in the file? It certainly doesn't work, at least in chrome (macOS & Windows)

**Comment by clementroche** [comment]
yes but i don't see what's wrong with this implementation

**Comment by clementroche** [comment]
yes thanks, so ideally you'll need ref that expose `{lenis,wrapperRef,contentRef}` instead of just `{lenis}`

**Comment by damnsamn** [comment]
@clementroche
I'll attach a screen recording... Are you not getting the same behaviour? I'm able to reproduce the issue on multiple machines via that link
https://github.com/studio-freight/react-lenis/assets/17864782/17bd16fe-db59-494d-b909-a058cb950a20

**Comment by clementroche** [comment]
read my comment again

**Comment by damnsamn** [comment]
Are you able to produce a functioning example? I must be missing something...

> so ideally you'll need ref that expose {lenis,wrapperRef,contentRef} instead of just {lenis}

Yes I think a ref that exposes the element ref directly would probably work?

**Comment by clementroche** [comment]
fixed in v0.0.40
